### PR TITLE
Alt click and flashlight

### PR DESCRIPTION
**Pull request by lbnesquik**

### CODE REVIEW

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -100,34 +100,71 @@
 	var/ammo_x_offset = 2
 	var/ammo_y_offset = 0
 	var/can_flashlight = 0 //This means "Can we add a light to this"
-	var/obj/item/device/flashlight/F = null //added by aeiou
-	var/gun_light = FALSE //Is there a light
-	var/light_state = "flight"
-	var/light_brightness = 4
+	var/obj/item/device/flashlight/flght = null //The actual light attachement.
+	var/gun_light = 0 //is the light on(?)
+	var/light_state = "flght"
```
Comment (EvilJackCarver) [inline code/modules/projectiles/gun.dm]: this is the actual state name for the overlay, not the variable name referencing the actual flashlight itself

File: code/game/objects/items/devices/geiger.dm [hunk 2]
```diff
@@ -54,10 +54,16 @@
...
 /obj/item/device/geiger/attack_self(var/mob/user)
+	toggle()
+
+/obj/item/device/geiger/proc/toggle(var/mob/user)//Moved here to easy the alt click process
 	scanning = !scanning
 	update_icon()
 	to_chat(user, "<span class='notice'>\icon[src] You switch [scanning ? "on" : "off"] \the [src].</span>")
 
+/obj/item/device/geiger/AltClick()//AEIOU EDIT
+	toggle()
```
Comment (EvilJackCarver) [inline code/game/objects/items/devices/geiger.dm]: it'd be a good idea to double-check that ghosts can't turn on and off the geiger counter this way

File: code/game/mecha/mecha_construction_paths.dm [hunk 3]
```diff
@@ -332,19 +332,19 @@
...
+					 		"desc"="Capacitor is secured"),
 					 //7
 					 list("key"=/obj/item/weapon/screwdriver,
 					 		"backkey"=/obj/item/weapon/crowbar,
-					 		"desc"="Advanced capacitor is installed"),
+					 		"desc"="Capacitor is installed"),
 					 //8
 					 list("key"=/obj/item/weapon/stock_parts/capacitor/adv,
 					 		"backkey"=/obj/item/weapon/screwdriver,
-					 		"desc"="Advanced scanner module is secured"),
```
Comment (drexample) [inline code/game/mecha/mecha_construction_paths.dm]: Why tho.

File: code/modules/projectiles/gun.dm [hunk 4]
```diff
@@ -100,34 +100,71 @@
...
-/obj/item/weapon/gun/CtrlClick(mob/user)
+//////////////////////////////////////
+/obj/item/weapon/gun/AltClick(mob/user)
 	if(can_flashlight && ishuman(user) && src.loc == usr && !user.incapacitated(INCAPACITATION_ALL))
 		toggle_flashlight()
 	else
 		return ..()
 
-/obj/item/weapon/gun/proc/toggle_flashlight()
+/obj/item/weapon/gun/proc/toggle_flashlight(user)
```
Comment (drexample) [inline code/modules/projectiles/gun.dm]: user is just a variable name, passing it just like that won't add a user, since you don't pass it previously

**Pull request closed without merging**